### PR TITLE
Bar sign maint/dismantle bug fix.

### LINKED ISSUE #14884: Bar Sign is invisible once removed
**Issue Description**:
The bar sign turns invisible when you remove it from the wall with a screwdriver

**What did you expect to happen**: 
To have a sign that was able to be moved elsewhere, or one that can't move.

**What happened instead**:
Sign turned invisible, was still interactable via ALT-clicking the tile where it was, and interacting with it via the "floor" tab. When attempting to place it back on a wall, it remained invisible.

**Why is this bad/What are the consequences**:
Invisible items are bad, plus it would be nice to be able to move the bar sign around for bar rearranging.

**Steps to reproduce the problem**:
Unscrew sign with screwdriver.

### PR DESCRIPTION
**Pull request by Tuulipommi**
## What Does This PR Do

Removed the ability to unscrew and dismantle
the bar sign, as it caused the bar sign to
disappear, without a way to make it visible again.
This was done, as there were no sprites for either of these functions.
fixes #14884
## Why It's Good For The Game

As stated in the [issue report](https://github.com/ParadiseSS13/Paradise/issues/14884), invisible objects aren't really a good addition to the game. Yes, its a cosmetic one, but it can be a bit distracting.
## Changelog
:cl: Windbomb
fix: you can no longer break the bar sign by trying to disassemble it
/:cl:

### CODE REVIEW

File: code/game/objects/structures/barsign.dm
Comment (FlattestGuitar) [inline code/game/objects/structures/barsign.dm:59]: Add a `/` at the start of this line.

**Comment by FlattestGuitar** [comment]
The changelog is for the players. They don't care about how many pieces of code you added or removed. 

How about `fix: you can no longer break the bar sign by trying to disassemble it`?

File: code/game/objects/structures/barsign.dm
Comment (Tuulipommi) [reply code/game/objects/structures/barsign.dm]: Fixed

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:90]: This part of the code should be part of the `screwdriver_act` proc. You used to be able to screw the panel open and the barsign had special logic for it to avoid the weirdness we have now.

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:107]: This can all stay in the `attackby` proc since then you can repair the barsign

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:50]: Should remain here. This is called when you destroy the sign which makes it drop the items listed above

**Review by farie82 — CHANGES_REQUESTED** [review]
Be sure to also include the following to your PR description:
fixes #14884

This will ensure that once this PR gets merged the issue gets closed since it is fixed

File: code/game/objects/structures/barsign.dm
Comment (variableundefined) [inline code/game/objects/structures/barsign.dm:82]: Spacing

File: code/game/objects/structures/barsign.dm
Comment (variableundefined) [inline code/game/objects/structures/barsign.dm:99]: Implicit src, please remove

**Review by variableundefined — CHANGES_REQUESTED** [review]
Also fix the things the linter pointed out. Please do not webedit.

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:82]: This check should also not exist. `screwdriver_act` means an item was used with the property of being a screwdriver tool. The old way used istype checks

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:100]:
```suggestion
		to_chat(user, "<span class='info'>Access denied.</span>")
```
spacing in html string is done like this. In DM it's correct though to put a space in between

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:103]:
```suggestion
		to_chat(user, "<span class='danger'>The controls seem unresponsive.</span>")
```

File: code/game/objects/structures/barsign.dm
Comment (farie82) [inline code/game/objects/structures/barsign.dm:98]:
```suggestion
/obj/structure/sign/barsign/attack_hand(mob/user)
```
`as mob` does nothing here. It's also the old way of doing things. Nowadays just a `mob/user` is plenty

**Comment by variableundefined** [comment]
@Tuulipommi Since this is a small PR, I am requesting a fix within 4 days or the PR may be closed.

**Comment by Kiyahitayika** [comment]
@Tuulipommi First off, thank you for working on this and tis was a really good first PR.  Second, you are dealing with code that has not been touched in a while here.

Farie is right about the `as mob` lines. It looks like they've been there for a while - just `mob/user` is fine. The spacing in the HTML output was like that already and it's easy to fix as well. Cleaning up and tweaking old code is something that eventually becomes old-hat when working on the Paradise codebase. Contributors and Maintainers like to keep the code fresh and update it whenever anything is touched.

VariableUndefined is right about the implicit src. This happens a lot in older code. Instead of `if(!src.allowed(user))`, the code can nowadays use `if(!allowed(user))` because it recognizes the 'src' as the datum this line of code is being called on (barsign) and sees that the `allowed` proc is referring to the barsign. 
The `allowed` proc itself is passed all the way up from /obj/ through /structure/ through /signs/ to /barsign/ so we can ask "hey, does this `user` have access to this object?". Line 6 of this code file sets that access with `req_access = list(ACCESS_BAR)`.

There are three other instances of implied src in this file as well. Two are here:
```
		if(BRUTE)
			playsound(src.loc, 'sound/effects/glasshit.ogg', 75, TRUE)
		if(BURN)
			playsound(src.loc, 'sound/items/welder.ogg', 100, TRUE)

```
In both cases, `src.loc` can be `loc`. The third one is:
```
/obj/structure/sign/barsign/attack_ai(mob/user)
	return src.attack_hand(user)
```

All this to say - you are on the right path with this! I was able to get your code running with the changes suggested in the reviews (and a change in indentation). This looks like an excellent first PR and I don't want you to give up on coding if you're interested in it!

I am planning on using your work from this project to fix the same issue on a Paradise downstream server. Seriously, good work, and don't get discouraged. If you're not on the Paradise Discord, I recommend joining it and reaching out to Farie and joining the coding channel. Paradise is good people and coding is tough to do alone.

**Pull request closed without merging**
